### ISSUE TITLE
Setting duplication causing massive performance issues.

### ISSUE THREAD

**Issue by maodus** [PSP-Archive/ARK-4#373]
### PSP/Vita Model

PSP-3000 (Slim & Bright)

### ARK Version

ARK 4.20.69 r5

### Homebrew/GAME (if applicable)

N/A

### Plugins ( if applicable )

xmb, ms0:/SEPLUGINS/CXMB/CXMB.PRX, off
game, ms0:/SEPLUGINS/nploader_mod/nploader_mod.prx, on
xmb, ms0:/SEPLUGINS/npdrm_free_mod/npdrm_free_mod.prx, off
game, ms0:/SEPLUGINS/npdrm_free_mod/npdrm_free_mod.prx, off

### Steps to reproduce

1. Change a setting
2. Save the changed setting by hitting X

In doing so, the SETTINGS.txt will duplicate the line 

```
# Luxor doesn't like Inferno Cache
ULUS10201, infernocache, off
```

This will eventually lead to absurd wait times for when settings are read. i.e. booting, saving settings, launching games. Ive attached my SETTINGS.txt so please refer to that.
[SETTINGS.TXT](https://github.com/user-attachments/files/15744452/SETTINGS.TXT)

### USB Charge

Disabled

### Overclock

Disabled

### Power Save

Disabled

### Balanced Energy Mode

Launcher

### Autoboot Launcher

On

### Disable PSP Go Pause

Off

### Unlock Extra Memory

Disabled

### Memory Stick Speedup

Disabled

### Inferno Cache

Off

### Skip Sony Logos

On

### Hide PIC0 and PIC1

Off

### Hide MAC Address

On

### Hide DLC

On

### Turn off LEDs

Disabled

### Verification of form.

- [X] I agree that I properly filled out the items listed above.

**Comment by krazynez** [comment]
I cannot reproduce this. @JoseAaronLopezGarcia can you try?

**Comment by maodus** [comment]
Yeah im not sure if a specific setting caused this to start happening, nor do I know when this started happening. I will say i was changing a lot of the setting quite frequently to see what they did. But I never edited/malformed the SETTINGS.txt directly.

**Comment by krazynez** [comment]
> Yeah im not sure if a specific setting caused this to start happening, nor do I know when this started happening. I will say i was changing a lot of the setting quite frequently to see what they did. But I never edited/malformed the SETTINGS.txt directly.

Good to know that was my next question actually (manually or not editing them)

**Comment by maodus** [comment]
I can still actively reproduce this even when leaving only a single 
```
# Luxor doesn't like Inferno Cache
ULUS10201, infernocache, off
```

In the SETTINGS.txt. So lmk if you need any other files from my system that might correlate.

**Comment by krazynez** [comment]
> I can still actively reproduce this even when leaving only a single
> 
> ```
> # Luxor doesn't like Inferno Cache
> ULUS10201, infernocache, off
> ```
> 
> In the SETTINGS.txt. So lmk if you need any other files from my system that might correlate.

how about remove the SETTINGS.TXT completely then also upgrade to r7 (just because its the latest anyways) then try to change settings and see what happens

**Comment by maodus** [comment]
> > I can still actively reproduce this even when leaving only a single
> > ```
> > # Luxor doesn't like Inferno Cache
> > ULUS10201, infernocache, off
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In the SETTINGS.txt. So lmk if you need any other files from my system that might correlate.
> 
> how about remove the SETTINGS.TXT completely then also upgrade to r7 (just because its the latest anyways) then try to change settings and see what happens

Removing SETTINGS.txt fixed it. When the new SETTINGS.txt was created. The problematic inferno cache line wasn't there by default. So maybe its only an issue with the settings file that is shipped by default? Specifically that inferno cache line.

**Comment by maodus** [comment]
I figured out what I was doing and how to reproduce it even on r7.

1. Start with clean SETTINGS.txt
2. insert the lines # Luxor doesn't like Inferno Cache ULUS10201, infernocache, off
3. Hit SELECT in games section to ensure the new lines were loaded
4. Change a few settings and save them
5. Restart the system
6. Change a few more settings
7. Restart again

You will now have duplicated the custom config once or twice. I always restart my system whenever i make changes just to ensure everything works and i guess that is where the bug lies. Your custom config gets messed up/isn't interpreted correctly when you restart your system immediately after making changes to your settings.

**Comment by maodus** [comment]
Fixed with dd37a3a.
